src/DIRAC/WorkloadManagementSystem/Executor/Base/OptimizerExecutor.py

```diff
@@ -72,7 +44,7 @@ def initializeOptimizer(cls):
 
     def processTask(self, jid, jobState):
         self.__jobData.jobState = jobState
-        self.__jobData.jobLog = self.JobLog(self.log, jid)
+        self.__jobData.jobLog = gLogger.getSubLogger(f"[JID {jid}]{self.__class__.__name__}")
```

Comment:
Well right now the logs look like
```
WorkloadManagement/Optimizers/WorkloadManagement/JobScheduling INFO: [JID 123] the_actual_message
```
while with this change they will look like
```
INFO: [JID 123]JobScheduling the_actual_message
```
which does not look better. 

I suggest to try with:
```suggestion
        self.__jobData.jobLog = self.log.getSubLogger(f"[JID {jid}]")
```

And see in your local setup.